src/puppetlabs/comidi.clj

```diff
@@ -63,9 +70,9 @@
       (str/replace #"-$" "")))
 
 (defn special-chars->underscores
-  "Convert all non-alpha chars except * and - to underscores"
+  "Convert all non-alpha chars except ! * and - to underscores"
   [s]
-  (str/replace s #"[^\w\*\-]" "_"))
+  (str/replace s #"[^\w\!\*\-]" "_"))
```

Comment:
nm, kinda, on this comment :)  I now see a bit more about what's causing this w/rt the `false` stuff below, but I don't entirely get it.